From: Alexandre Esse <alexandre.esse@canonical.com>
Date: Fri, 7 Jun 2024 13:19:26 +0200
Subject: cmake: use PROGRAMS form for signature.sh
Forwarded: https://github.com/Elektrobit/samconf/pull/2

The PROGRAMS form is identical to the FILES form except that the default
permissions for the installed file also include OWNER_EXECUTE, GROUP_EXECUTE,
and WORLD_EXECUTE. This form is intended to install programs that are not
targets, such as shell scripts.

Signed-off-by: Alexandre Esse <alexandre.esse@canonical.com>
---
 src/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 3d190c1..e91135f 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -8,7 +8,7 @@ endif (SAMCONF_DEMOS)
 add_subdirectory(samconf)
 
 if (INSTALL_SAMCONF_TOOLS)
-  install(FILES ${PROJECT_SOURCE_DIR}/src/signature.sh
-          PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_EXECUTE WORLD_EXECUTE
+  install(PROGRAMS ${PROJECT_SOURCE_DIR}/src/signature.sh
+          PERMISSIONS OWNER_READ OWNER_WRITE
           DESTINATION ${CMAKE_INSTALL_BINDIR})
 endif (INSTALL_SAMCONF_TOOLS)
